glambie/processing/processing_helpers.py

```diff
@@ -345,6 +353,64 @@ def slice_timeseries_at_gaps(df_timeseries: pd.DataFrame) -> list[pd.DataFrame]:
     return split_timeseries_dataframes
 
 
+def recombine_split_timeseries_in_catalogue(data_catalogue: DataCatalogue,
+                                            names_of_split_datasets_in_catalogue: list[list]) -> DataCatalogue:
+    """
+    Combines a list of split timeseries back into full timeseries
+    Can be used after gaps have been removed
+
+    Parameters
+    ----------
+    data_catalogue : DataCatalogue
+        input data catalogue which potentially have been split up due to gaps
+    names_of_split_datasets_in_catalogue : list[list]
+        list with user group names of datasets within the data catalogue that have been split
+        e.g. [["rabbit_1", "rabbit_2"], ["seal_1", "seal_2", "seal_3"]]
+
+    Returns
+    -------
+    DataCatalogue
+        data catalogue with timeseries that are now recombined
+    """
+    new_datasets = []
+    # split datasets
+    for split_ds_list in names_of_split_datasets_in_catalogue:
+        old_datasets = [data_catalogue.get_filtered_catalogue(user_group=s).datasets[0] for s in split_ds_list]
+        start_dates = []
+        end_dates = []
+        changes = []
+        errors = []
+        for ds in old_datasets:
+            start_dates.extend(ds.data.start_dates)
+            end_dates.extend(ds.data.end_dates)
+            changes.extend(ds.data.changes)
+            errors.extend(ds.data.errors)
+        df = pd.DataFrame({"start_dates": start_dates, "end_dates": end_dates,
+                           "changes": changes, "errors": errors})
+        df = df.sort_values(by="start_dates")
+        if len(start_dates) != len(np.unique(start_dates)) or len(end_dates) != len(np.unique(end_dates)):
+            error_msg = f'''Issue with combining split datasets, duplicate dates discovered when combining:
+            {split_ds_list}'''
+            log.error(error_msg)
+            raise ValueError(error_msg)
+        new_dataset = old_datasets[0].copy()
```

Comment:
Could you take the 0 index or use `next` when you define it on line 378? i.e.
```old_datasets = [data_catalogue.get_filtered_catalogue(user_group=s).datasets[0] for s in split_ds_list][0]```
or
```old_datasets = next([data_catalogue.get_filtered_catalogue(user_group=s).datasets[0] for s in split_ds_list])``` :D

Reply:
This line is just used to copy the metadata of the first split dataset to fill the matadata of the newly combined dataset. We just take the first one as they should all have the same metadata, so no need to iterate to the other elements.
I realise this is not clear from the code and added a comment above to explain :)

Reply:
*metadata

Reply:
Lovely, thanks!